ale_linters/prolog/swipl.vim

```diff
@@ -34,13 +34,14 @@ function! s:Subst(format, vars) abort
     return substitute(a:format, '%\(.\)', '\=get(l:vars, submatch(1), "")', 'g')
 endfunction
 
+let s:pattern = '\v^(ERROR|Warning)+%(:\s*[^:]+:(\d+)%(:(\d+))?)?:\s*(.*)$'
```

Comment:
Revert this back to l:pattern and move it back inside the handle function. No need to make this a script scoped variable.